### ISSUE TITLE
$in query causing ClassCastException: java.util.ArrayList cannot be cast to java.lang.Comparable

### ISSUE THREAD

**Issue by runtologist** [bwaldvogel/mongo-java-server#57]
A query using `$in` on the `_id` field fails with a ClassCastException as shown in the logs below. Matching on other fields works as expected.

```
15:27:40.187 [mongo-server-worker2] DEBUG d.b.m.w.MongoWireProtocolHandler - MongoQuery(header: MessageHeader(request: 112, responseTo: 0), collection: xxx.xxx, query: {"_id" : {"$in" : [[1, 2, 3]]}}, returnFieldSelector: null)
15:27:40.192 [mongo-server-worker2] ERROR d.b.m.w.MongoExceptionHandler - exception for client 5f418d5e
java.lang.ClassCastException: java.util.ArrayList cannot be cast to java.lang.Comparable
	at java.util.TreeMap.compare(TreeMap.java:1294)
	at java.util.TreeMap.put(TreeMap.java:538)
	at java.util.TreeSet.add(TreeSet.java:255)
	at java.util.AbstractCollection.addAll(AbstractCollection.java:344)
	at java.util.TreeSet.addAll(TreeSet.java:312)
	at java.util.TreeSet.<init>(TreeSet.java:160)
	at de.bwaldvogel.mongo.backend.AbstractUniqueIndex.getPositionsForExpression(AbstractUniqueIndex.java:170)
	at de.bwaldvogel.mongo.backend.AbstractUniqueIndex.getPositions(AbstractUniqueIndex.java:135)
	at de.bwaldvogel.mongo.backend.AbstractMongoCollection.queryDocuments(AbstractMongoCollection.java:50)
	at de.bwaldvogel.mongo.backend.AbstractMongoCollection.handleQuery(AbstractMongoCollection.java:718)
	at de.bwaldvogel.mongo.backend.AbstractMongoDatabase.handleQuery(AbstractMongoDatabase.java:650)
	at de.bwaldvogel.mongo.backend.AbstractMongoBackend.handleQuery(AbstractMongoBackend.java:206)
	at de.bwaldvogel.mongo.wire.MongoDatabaseHandler.handleQuery(MongoDatabaseHandler.java:93)
	at de.bwaldvogel.mongo.wire.MongoDatabaseHandler.channelRead0(MongoDatabaseHandler.java:71)
	at de.bwaldvogel.mongo.wire.MongoDatabaseHandler.channelRead0(MongoDatabaseHandler.java:37)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:677)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:612)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:529)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:491)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:905)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by bwaldvogel** [comment]
Thanks for the bug report.
I already started working on the fix but along the way found more cases that need to be fixed…

**Comment by bwaldvogel** [comment]
It should be fixed with 3d3734c.
Can you retest?

**Comment by bwaldvogel** [comment]
The fix is contained in version `1.14.0`.

Please reopen or create a new issue if you it still doesn’t work for you.

**Comment by runtologist** [comment]
Works like a charm. Thanks a lot. :)
